babel/babel
# Rewrite `transform-block-scoping` plugin

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #15171
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT



This PR rewrites the `transform-block-scoping` plugin, to:
- make it easier to follow what's going on, avoiding the stateful `BlockScoping` class and just relying on plain visitors
- clearly separate Annex B.3.3 behavior
- make the transform-block-scoping behavior consistent regardless of the context (function scope vs global scope; for vs while; loop scope vs block scope inside loop)

Lets see if it breaks any test262 tests

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15200"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
